src/h5web/vis-packs/core/heatmap/utils.ts

```diff
@@ -14,15 +14,17 @@ const GRADIENT_PRECISION = 1 / 20;
 const GRADIENT_RANGE = range(0, 1 + GRADIENT_PRECISION, GRADIENT_PRECISION);
 
 export function getVisDomain(
-  dataDomain: Domain,
   customDomain: CustomDomain,
+  dataDomain: Domain
+): Domain {
+  return [customDomain[0] ?? dataDomain[0], customDomain[1] ?? dataDomain[1]];
+}
+
+export function getSafeDomain(
+  visDomain: Domain,
+  dataDomain: Domain,
```

Comment:
`dataDomain` could be renamed to `fallbackDomain`

Reply:
and `visDomain` to just `domain`